DSP-1033 Reposition List Nodes

SepidehAlassi opened the pull request:
resolves DSP-1033
new route for this: PUT `admin/lists/<nodeIri>/position`. Requires `parentIri` and the new `position` in the request body.
- [x] reposition a child node within the same parent, shift siblings accordingly.
- [x] change parent of a child node, and position it in a specific position among new siblings. Shift the new siblings accordingly.
- [x] if position=-1, move the node to the end of children of  the given parent node.
- [x] add tests 
- [x] add test data
- [x] add documentation

benjamingeer commented on webapi/src/main/scala/org/knora/webapi/responders/admin/ListsResponderADM.scala:
```diff
@@ -1181,6 +1182,276 @@ class ListsResponderADM(responderData: ResponderData) extends Responder(responde
+            // If the node must be added to a new parent, highest valid position is numberOfChildren+1
+            // That means the furthests a node can be positioned is being appended to the end of children of the new parent.
+            if(isNewParent && changeNodePositionRequest.position >  numberOfChildren + 1) {
+                throw BadRequestException(s"Invalid position given, maximum allowed is=${numberOfChildren + 1}!")
```
Maybe this is just me, but I don't like receiving error messages with exclamation marks in them, it makes me feel like the computer is shouting at me. :)

benjamingeer commented on webapi/src/main/scala/org/knora/webapi/responders/admin/ListsResponderADM.scala:
```diff
@@ -1181,6 +1182,276 @@ class ListsResponderADM(responderData: ResponderData) extends Responder(responde
+            }
+        }
+        /**
+         * Checks that the position of the node is updated and node is sublist of specified parent.
```
Please finish the `@throws` comment.

benjamingeer reviewed (approved):
This looks great. Please try reformatting these files using the [dsp-intellij-settings](https://github.com/dasch-swiss/dsp-intellij-settings) repo that I've just made.

benjamingeer commented:
Did I review this by mistake? I thought I had got a review request but maybe I didn't. :)

benjamingeer reviewed (changes requested):
This is even better than the first time I reviewed it. :) But please reformat it with the shared IntelliJ settings.

SepidehAlassi commented:
> This is even better than the first time I reviewed it. :) But please reformat it with the shared IntelliJ settings.

I have reformatted it already :/

benjamingeer commented:
> I have reformatted it already :/

I've just done commit b056bb2 so you can see what `Reformat Code` does in my IntelliJ with that config.

benjamingeer commented:
Hmm but I don't like what it did here:

```scala
            sparqlQuery <- Future(org.knora.webapi.messages.twirl.queries.sparql.admin.txt.getListNode(
                triplestore = settings.triplestoreType,
                nodeIri = nodeIri
            ).toString())
            ).toString()
            )
```

I wanted it to look like this:

```scala
            sparqlQuery <- Future(
                org.knora.webapi.messages.twirl.queries.sparql.admin.txt.getListNode(
                    triplestore = settings.triplestoreType,
                    nodeIri = nodeIri
                ).toString()
            )
```

benjamingeer reviewed (approved):
I've fixed the formatting in `ListsResponderADM` manually, and changed the settings repository to hopefully produce better results.

SepidehAlassi commented:
> I've fixed the formatting in `ListsResponderADM` manually, and changed the settings repository to hopefully produce better results.

thank you very much for reformatting the code and for reviewing this PR.

subotic commented:
@SepidehAlassi @benjamingeer I think that it should be possible to store all settings inside https://github.com/dasch-swiss/dsp-api/blob/main/.editorconfig. The EditorConfig plugin comes bundled with IntelliJ. This would also allow, other IDEs to be configured correctly.

subotic reviewed (approved):
Thanks, looks great!
